src/bin/asr_mc_train.py

```diff
@@ -0,0 +1,200 @@
+                        help='Apply label smoothing with a specified distribution type')
+    parser.add_argument('--lsm-weight', default=0.0, type=float,
+                        help='Label smoothing weight')
+    # beamformer
+    parser.add_argument('--btype', default='blstmp', type=str,
+                        choices=['blstmp'],
+                        help='Type of beamformer network architecture')
+    parser.add_argument('--blayers', default=3, type=int,
+                        help='Number of beamformer layers')
+    parser.add_argument('--bunits', default=300, type=int,
+                        help='Number of beamformer hidden units')
+    parser.add_argument('--bprojs', default=320, type=int,
+                        help='Number of beamformer projection units')
+    parser.add_argument('--badim', default=320, type=int,
+                        help='Number of attention transformation dimensions in beamformer')
+    # model (parameter) related
+    parser.add_argument('--dropout-rate', default=0.0, type=float,
+                        help='Dropout rate')
+    # minibatch related
+    parser.add_argument('--batch-size', '-b', default=50, type=int,
+                        help='Batch size')
+    parser.add_argument('--maxlen-in', default=800, type=int, metavar='ML',
+                        help='Batch size is reduced if the input sequence length > ML')
+    parser.add_argument('--maxlen-out', default=150, type=int, metavar='ML',
+                        help='Batch size is reduced if the output sequence length > ML')
+    # optimization related
+    parser.add_argument('--opt', default='adadelta', type=str,
+                        choices=['adadelta', 'adam'],
+                        help='Optimizer')
+    parser.add_argument('--eps', default=1e-8, type=float,
+                        help='Epsilon constant for optimizer')
+    parser.add_argument('--eps-decay', default=0.01, type=float,
+                        help='Decaying ratio of epsilon')
+    parser.add_argument('--criterion', default='acc', type=str,
+                        choices=['loss', 'acc'],
+                        help='Criterion to perform epsilon decay')
+    parser.add_argument('--threshold', default=1e-4, type=float,
+                        help='Threshold to stop iteration')
+    parser.add_argument('--epochs', '-e', default=30, type=int,
+                        help='Number of maximum epochs')
+    parser.add_argument('--grad-clip', default=5, type=float,
+                        help='Gradient norm threshold to clip')
+    # me2e
+    parser.add_argument('--melmat', type=str, required=True,
+                        help='Filename of Mel-filterbank matrix data (Kaldi ark)')
+    parser.add_argument('--cmvn', type=str, required=True,
+                        help='Filename of cmvn statistics data (Kaldi ark)')
+    parser.add_argument('--mode', default='noisy+enhan', type=str,
```

Comment:
`--mode` is not informative name. maybe `--utt-type` is better? @sw005320 @TsubasaOchiai

Reply:
Now, I wonder whether I remove this option or not.
Because this option is made for experimental purpose, I think it is not needed for a stable version...
I would like to know your opinion.

Reply:
I always prefer simpler codes. If you think so, remove them.

Reply:
Thank you for your opinion.
I will remove the option later.